ceph/rados/serviceability_workflows.py

```diff
@@ -227,117 +196,40 @@ def remove_custom_host(self, host_node_name: str):
-                      case1: If the ceph is still in process of removing the OSD the command generated
-                      the proper json output.The json.loads method loads the output without any failure.
-                      case2: If the OSDs are removed from the node then the command wont generate any output.
-                      In this case the json.loads method throws the JSONDecodeError exception.This is the
-                      confirmation that the OSDs removal are completed. """
-                end_time = datetime.datetime.now() + datetime.timedelta(seconds=600)
-                log.debug(f"{txt_osd_removed_logic}")
-                while end_time > datetime.datetime.now():
-                    out, err = self.cephadm.shell([status_cmd])
-                    try:
-                        drain_ops = json.loads(out)
-                        for entry in drain_ops:
-                            log.debug(
-                                f"OSD remove operation is in progress {osd_id}\nOperations: {entry}"
-                            )
-                    except json.JSONDecodeError:
-                        log.info(f"The OSD removal is completed on OSD : {osd_id}")
-                        status_flag = True
-                        break
-                    except Exception as error:
-                        log.error(f"Hit issue during drain operations: {error}")
-                        raise Exception(error)
-                    log.debug("Sleeping for 10 seconds and checking again....")
-                    time.sleep(10)
-                return status_flag
-
-            daemon_check = self.rados_obj.check_daemon_exists_on_host(
-                host=host_node_name, daemon_type=None
-            )
-
-            if not daemon_check:
-                log.info(f" The node {host_node_name} is already drained.")
-                return None
             # Removing an OSD host and checking status
             rm_host = utils.get_node_by_id(self.cluster, host_node_name)
             log.info(
                 f"Identified host : {rm_host.hostname} to be removed from the cluster"
             )
 
-            # Get list of osd_id on the host to be removed
+            # get list of osd_id on the host to be removed
             rm_osd_list = self.rados_obj.collect_osd_daemon_ids(osd_node=rm_host)
             log.info(
-                f"The osd id  list to be removed from the {rm_host} is  {rm_osd_list}"
+                "The OSD list to be removed from the %s is %s"
+                % (rm_host.hostname, rm_osd_list)
             )
-            # Get the OSD out list and remove before drain the node
-            osd_out_list = self.rados_obj.get_osd_list(status="out")
-            log.info(
-                f"The out osd id  list to be removed from the {rm_host} is  {osd_out_list}"
+
+            daemon_check = self.rados_obj.check_daemon_exists_on_host(
```

Comment:
what does `daemon_type=None` collect? all the daemons?

Reply:
Add another check here:  Check if `_no_schedule ` label is added to host.  This will tell us if the drain command was previously run on the host.

Reply:
Yes, daemon_type=None collects all the daemons, if no daemons are found, the method returns False

Label check for `_no_schedule` has been added